ui/modal/hex_wallet_restore_modal.go

```diff
@@ -0,0 +1,238 @@
+	hm.positiveButtonText = text
+	hm.positiveButtonClicked = clicked
+	return hm
+}
+
+func (hm *HexRestoreModal) NegativeButton(text string, clicked func()) *HexRestoreModal {
+	hm.negativeButtonText = text
+	hm.negativeButtonClicked = clicked
+	return hm
+}
+
+func (hm *HexRestoreModal) SetLoading(loading bool) {
+	hm.isLoading = loading
+	hm.modal.SetDisabled(loading)
+}
+
+func (hm *HexRestoreModal) SetCancelable(min bool) *HexRestoreModal {
+	hm.isCancelable = min
+	return hm
+}
+
+func (hm *HexRestoreModal) SetError(err string) {
+	if err == "" {
+		hm.hexEditor.ClearError()
+	} else {
+		hm.hexEditor.SetError(err)
+	}
+}
+
+func (hm *HexRestoreModal) Handle() {
+	isSubmit, isChanged := decredmaterial.HandleEditorEvents(hm.hexEditor.Editor)
+	if isChanged {
+		hm.hexEditor.SetError("")
+		hex := hm.hexEditor.Editor.Text()
+		if len(hex) >= 16 && len(hex) <= 64 {
+			hm.btnPositve.SetEnabled(true)
+		} else {
+			hm.btnPositve.SetEnabled(false)
+		}
+	}
+
+	if hm.btnPositve.Button.Clicked() || isSubmit {
+		if !editorsNotEmpty(hm.hexEditor.Editor) {
+			hm.hexEditor.SetError("Enter Hex")
+			return
+		}
+
+		if hm.isLoading {
+			return
+		}
+
+		hm.SetLoading(true)
+		hm.SetError("")
+		if hm.positiveButtonClicked(hm.hexEditor.Editor.Text(), hm) {
+			hm.DismissModal(hm)
+		}
+	}
+
+	hm.btnNegative.SetEnabled(!hm.isLoading)
+	for hm.btnNegative.Clicked() {
+		if !hm.isLoading {
+			hm.DismissModal(hm)
+			hm.negativeButtonClicked()
+		}
+	}
+
+	if hm.modal.BackdropClicked(hm.isCancelable) {
+		if !hm.isLoading {
+			hm.Dismiss()
+			hm.negativeButtonClicked()
+		}
+	}
+}
+
+func (hm *HexRestoreModal) Layout(gtx layout.Context) D {
+	title := func(gtx C) D {
+		t := hm.Theme.H6(hm.dialogTitle)
+		t.Font.Weight = text.SemiBold
+		return t.Layout(gtx)
+	}
+
+	description := func(gtx C) D {
+		t := hm.Theme.Body2(hm.description)
+		return t.Layout(gtx)
+	}
+
+	editor := func(gtx C) D {
+		return hm.hexEditor.Layout(gtx)
+	}
+
+	actionButtons := func(gtx C) D {
+		return layout.E.Layout(gtx, func(gtx C) D {
+			return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					if hm.negativeButtonText == "" || hm.isLoading {
+						return layout.Dimensions{}
+					}
+
+					hm.btnNegative.Text = hm.negativeButtonText
+					return hm.btnNegative.Layout(gtx)
+				}),
+				layout.Rigid(func(gtx C) D {
+					if hm.isLoading {
+						return hm.materialLoader.Layout(gtx)
+					}
+
+					if hm.positiveButtonText == "" {
+						return layout.Dimensions{}
+					}
+
+					hm.btnPositve.Text = hm.positiveButtonText
+					return hm.btnPositve.Layout(gtx)
+				}),
+			)
+		})
+	}
+	var w []layout.Widget
+
+	w = append(w, title)
+
+	if hm.description != "" {
+		w = append(w, description)
+	}
+
+	if hm.customWidget != nil {
+		w = append(w, hm.customWidget)
+	}
+
+	w = append(w, editor)
+	w = append(w, actionButtons)
+
+	return hm.modal.Layout(gtx, w)
```

Comment:
We dont need this modal. We already have the `text input modal` to handle scenarios of this nature.